CMSgov/easi-app
# [EASI-3730] Business case lifecycle costs bug

# EASI-3730

## Changes and Description

This is part of the UX findings for IT Gov v2 ([Send business case button not working on final biz case](https://docs.google.com/document/d/1oZBAphS7VgoFIjcTXMSyXfb6LbAQMylTqPxMxlzHnyU/edit#heading=h.p5vodtcujsfu)).

Lifecycle costs are submitting as `null` if not filled out, which is triggering the backend validation to require those fields and causing a submission error.

### Frontend changes:
- Updated code in src/data/businessCase.ts to only submit lifecycle costs that have been filled out
- Field validation updates (confirmed with Zoe)
  - All cost fields are required for both the Development and Operations and Maintenance categories. The requester can enter 0 for all fields if there are no associated costs, but they cannot be blank.
  - If the requester adds a related cost from the dropdown, they have to either manually remove it using the “remove category” button or fill in all costs like the first two categories. 
  - Alternative A is NOT required for either draft or final business case



## How to test this change

1. Create intake request
2. Use "Progress to new step" v2 action to progress to draft business case
3. Fill out and submit draft business case with empty lifecycle cost fields
4. Use "Progress to new step" v2 action to progress to final business case
5. Fill out and submit final business case
    - Make sure lifecycle costs validation works as expected
    - Alternative A should **not** be required

**Note:** only submit the final business case once when testing - don't submit, request edits, and then try testing with the same final biz case. The v1/v2 statuses are mismatched, so if you try to resubmit a final business case it treats it like a draft and doesn't trigger any validation. This will be fixed in EASI-3440.



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
